plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -94,7 +113,12 @@ func (g *genericScheduler) Schedule(pod *api.Pod, nodeLister algorithm.NodeListe
 		}
 	}
 
-	return g.selectHost(priorityList)
+	selectNode, err := g.selectHost(priorityList)
+	// If binding pod to a node, invalid cache of this node.
+	if err == nil {
+		g.cache.InvalidNodeCache(selectNode)
```

Comment:
And we should watch ReplicationController and Service, too, if they updated, we should invalidate the according cache, since we `CalculateSpreadPriority` and `CalculateAntiAffinityPriority` in scheduler. when rc and service updated, these two priority functions may return different result.